### ISSUE TITLE
Emacs 29.0.60 "Invalid search bound (wrong side of point)"

### ISSUE THREAD

**Issue by alexeyre** [yangsheng6810/org-latex-instant-preview#17]
Weird error I'm getting on emacs 29.0.60 and org 9.6, causes the whole screen to flash/redisplay but the preview does display correctly.
```
⛔ Warning (org-element-cache): org-element--cache: Org parser error in main.org::86. Resetting.
 The error was: (error "Invalid search bound (wrong side of point)")
 Backtrace:
"  backtrace-to-string(nil)
  org-element-at-point()
  org--collect-keywords-1((\"SETUPFILE\" \"LATEX_CLASS\" \"LATEX_CLASS_OPTIONS\" \"LATEX_HEADER\" \"LATEX_HEADER_EXTRA\" \"DESCRIPTION\" \"KEYWORDS\" \"SUBTITLE\" \"LATEX_ENGRAVED_THEME\" \"LATEX_COMPILER\" \"DATE\" \"TITLE\" \"DATE\" \"AUTHOR\" \"EMAIL\" \"LANGUAGE\" \"SELECT_TAGS\" \"EXCLUDE_TAGS\" \"CREATOR\" \"CITE_EXPORT\" \"FILETAGS\" \"OPTIONS\") nil nil (\"/Users/alex/Documents/knowledge/main.org\") nil)
  org-collect-keywords((\"LATEX_CLASS\" \"LATEX_CLASS_OPTIONS\" \"LATEX_HEADER\" \"LATEX_HEADER_EXTRA\" \"DESCRIPTION\" \"KEYWORDS\" \"SUBTITLE\" \"LATEX_ENGRAVED_THEME\" \"LATEX_COMPILER\" \"DATE\" \"TITLE\" \"DATE\" \"AUTHOR\" \"EMAIL\" \"LANGUAGE\" \"SELECT_TAGS\" \"EXCLUDE_TAGS\" \"CREATOR\" \"CITE_EXPORT\" \"FILETAGS\" \"OPTIONS\"))
  org-export--get-inbuffer-options(#s(org-export-backend :name latex :parent nil :transcoders ((bold . org-latex-bold) (center-block . org-latex-center-block) (clock . org-latex-clock) (code . org-latex-code) (drawer . org-latex-drawer) (dynamic-block . org-latex-dynamic-block) (entity . org-latex-entity) (example-block . org-latex-example-block) (export-block . org-latex-export-block) (export-snippet . org-latex-export-snippet) (fixed-width . org-latex-fixed-width) (footnote-definition . org-latex-footnote-definition) (footnote-reference . org-latex-footnote-reference) (headline . org-latex-headline) (horizontal-rule . org-latex-horizontal-rule) (inline-src-block . org-latex-inline-src-block) (inlinetask . org-latex-inlinetask) (italic . org-latex-italic) (item . org-latex-item) (keyword . org-latex-keyword) (latex-environment . org-latex-latex-environment) (latex-fragment . org-latex-latex-fragment) (line-break . org-latex-line-break) (link . org-latex-link) (node-property . org-latex-node-property) (paragraph . org-latex-paragraph) (plain-list . org-latex-plain-list) (plain-text . org-latex-plain-text) (planning . org-latex-planning) (property-drawer . org-latex-property-drawer) (quote-block . org-latex-quote-block) (radio-target . org-latex-radio-target) (section . org-latex-section) (special-block . org-latex-special-block) (src-block . org-latex-src-block) (statistics-cookie . org-latex-statistics-cookie) (strike-through . org-latex-strike-through) (subscript . org-latex-subscript) ...) :options ((:latex-class \"LATEX_CLASS\" nil org-latex-default-class t) (:latex-class-options \"LATEX_CLASS_OPTIONS\" nil nil t) (:latex-header \"LATEX_HEADER\" nil nil newline) (:latex-header-extra \"LATEX_HEADER_EXTRA\" nil nil newline) (:description \"DESCRIPTION\" nil nil parse) (:keywords \"KEYWORDS\" nil nil parse) (:subtitle \"SUBTITLE\" nil nil parse) (:latex-active-timestamp-format nil nil org-latex-active-timestamp-format) (:latex-caption-above nil nil org-latex-caption-above) (:latex-classes nil nil org-latex-classes) (:latex-default-figure-position nil nil org-latex-default-figure-position) (:latex-default-table-environment nil nil org-latex-default-table-environment) (:latex-default-quote-environment nil nil org-latex-default-quote-environment) (:latex-default-table-mode nil nil org-latex-default-table-mode) (:latex-diary-timestamp-format nil nil org-latex-diary-timestamp-format) (:latex-engraved-options nil nil org-latex-engraved-options) (:latex-engraved-preamble nil nil org-latex-engraved-preamble) (:latex-engraved-theme \"LATEX_ENGRAVED_THEME\" nil org-latex-engraved-theme) (:latex-footnote-defined-format nil nil org-latex-footnote-defined-format) (:latex-footnote-separator nil nil org-latex-footnote-separator) (:latex-format-drawer-function nil nil org-latex-format-drawer-function) (:latex-format-headline-function nil nil org-latex-format-headline-function) (:latex-format-inlinetask-function nil nil org-latex-format-inlinetask-function) (:latex-hyperref-template nil nil org-latex-hyperref-template t) (:latex-image-default-scale nil nil org-latex-image-default-scale) (:latex-image-default-height nil nil org-latex-image-default-height) (:latex-image-default-option nil nil org-latex-image-default-option) (:latex-image-default-width nil nil org-latex-image-default-width) (:latex-images-centered nil nil org-latex-images-centered) (:latex-inactive-timestamp-format nil nil org-latex-inactive-timestamp-format) (:latex-inline-image-rules nil nil org-latex-inline-image-rules) (:latex-link-with-unknown-path-format nil nil org-latex-link-with-unknown-path-format) (:latex-src-block-backend nil nil org-latex-src-block-backend) (:latex-listings-langs nil nil org-latex-listings-langs) (:latex-listings-options nil nil org-latex-listings-options) (:latex-listings-src-omit-language nil nil org-latex-listings-src-omit-language) (:latex-minted-langs nil nil org-latex-minted-langs) (:latex-minted-options nil nil org-latex-minted-options) ...) :filters ((:filter-options . org-latex-math-block-options-filter) (:filter-paragraph . org-latex-clean-invalid-line-breaks) (:filter-parse-tree org-latex-math-block-tree-filter org-latex-matrices-tree-filter org-latex-image-link-filter) (:filter-verse-block . org-latex-clean-invalid-line-breaks)) :blocks nil :menu (108 \"Export to LaTeX\" ((76 \"As LaTeX buffer\" org-latex-export-as-latex) (108 \"As LaTeX file\" org-latex-export-to-latex) (112 \"As PDF file\" org-latex-export-to-pdf) (111 \"As PDF file and open\" (lambda ... ...))))))
  org-export-get-environment(#s(org-export-backend :name latex :parent nil :transcoders ((bold . org-latex-bold) (center-block . org-latex-center-block) (clock . org-latex-clock) (code . org-latex-code) (drawer . org-latex-drawer) (dynamic-block . org-latex-dynamic-block) (entity . org-latex-entity) (example-block . org-latex-example-block) (export-block . org-latex-export-block) (export-snippet . org-latex-export-snippet) (fixed-width . org-latex-fixed-width) (footnote-definition . org-latex-footnote-definition) (footnote-reference . org-latex-footnote-reference) (headline . org-latex-headline) (horizontal-rule . org-latex-horizontal-rule) (inline-src-block . org-latex-inline-src-block) (inlinetask . org-latex-inlinetask) (italic . org-latex-italic) (item . org-latex-item) (keyword . org-latex-keyword) (latex-environment . org-latex-latex-environment) (latex-fragment . org-latex-latex-fragment) (line-break . org-latex-line-break) (link . org-latex-link) (node-property . org-latex-node-property) (paragraph . org-latex-paragraph) (plain-list . org-latex-plain-list) (plain-text . org-latex-plain-text) (planning . org-latex-planning) (property-drawer . org-latex-property-drawer) (quote-block . org-latex-quote-block) (radio-target . org-latex-radio-target) (section . org-latex-section) (special-block . org-latex-special-block) (src-block . org-latex-src-block) (statistics-cookie . org-latex-statistics-cookie) (strike-through . org-latex-strike-through) (subscript . org-latex-subscript) ...) :options ((:latex-class \"LATEX_CLASS\" nil org-latex-default-class t) (:latex-class-options \"LATEX_CLASS_OPTIONS\" nil nil t) (:latex-header \"LATEX_HEADER\" nil nil newline) (:latex-header-extra \"LATEX_HEADER_EXTRA\" nil nil newline) (:description \"DESCRIPTION\" nil nil parse) (:keywords \"KEYWORDS\" nil nil parse) (:subtitle \"SUBTITLE\" nil nil parse) (:latex-active-timestamp-format nil nil org-latex-active-timestamp-format) (:latex-caption-above nil nil org-latex-caption-above) (:latex-classes nil nil org-latex-classes) (:latex-default-figure-position nil nil org-latex-default-figure-position) (:latex-default-table-environment nil nil org-latex-default-table-environment) (:latex-default-quote-environment nil nil org-latex-default-quote-environment) (:latex-default-table-mode nil nil org-latex-default-table-mode) (:latex-diary-timestamp-format nil nil org-latex-diary-timestamp-format) (:latex-engraved-options nil nil org-latex-engraved-options) (:latex-engraved-preamble nil nil org-latex-engraved-preamble) (:latex-engraved-theme \"LATEX_ENGRAVED_THEME\" nil org-latex-engraved-theme) (:latex-footnote-defined-format nil nil org-latex-footnote-defined-format) (:latex-footnote-separator nil nil org-latex-footnote-separator) (:latex-format-drawer-function nil nil org-latex-format-drawer-function) (:latex-format-headline-function nil nil org-latex-format-headline-function) (:latex-format-inlinetask-function nil nil org-latex-format-inlinetask-function) (:latex-hyperref-template nil nil org-latex-hyperref-template t) (:latex-image-default-scale nil nil org-latex-image-default-scale) (:latex-image-default-height nil nil org-latex-image-default-height) (:latex-image-default-option nil nil org-latex-image-default-option) (:latex-image-default-width nil nil org-latex-image-default-width) (:latex-images-centered nil nil org-latex-images-centered) (:latex-inactive-timestamp-format nil nil org-latex-inactive-timestamp-format) (:latex-inline-image-rules nil nil org-latex-inline-image-rules) (:latex-link-with-unknown-path-format nil nil org-latex-link-with-unknown-path-format) (:latex-src-block-backend nil nil org-latex-src-block-backend) (:latex-listings-langs nil nil org-latex-listings-langs) (:latex-listings-options nil nil org-latex-listings-options) (:latex-listings-src-omit-language nil nil org-latex-listings-src-omit-language) (:latex-minted-langs nil nil org-latex-minted-langs) (:latex-minted-options nil nil org-latex-minted-options) ...) :filters ((:filter-options . org-latex-math-block-options-filter) (:filter-paragraph . org-latex-clean-invalid-line-breaks) (:filter-parse-tree org-latex-math-block-tree-filter org-latex-matrices-tree-filter org-latex-image-link-filter) (:filter-verse-block . org-latex-clean-invalid-line-breaks)) :blocks nil :menu (108 \"Export to LaTeX\" ((76 \"As LaTeX buffer\" org-latex-export-as-latex) (108 \"As LaTeX file\" org-latex-export-to-latex) (112 \"As PDF file\" org-latex-export-to-pdf) (111 \"As PDF file and open\" (lambda ... ...))))))
  org-latex-impatient--get-headers()
  org-latex-impatient-start()
  apply(org-latex-impatient-start nil)
  timer-event-handler([t 0 0 100000 nil org-latex-impatient-start nil idle 0 nil])
"
 Please report this to Org mode mailing list (M-x org-submit-bug-report).
```
Possible, even likely, that this is a configuration error on my part, but I wanted to submit in case it's not, so apologies if this is my fault...

**Comment by yangsheng6810** [comment]
Thanks for submitting the issue.

Looking at the backtrace, it seems the error comes from the following piece of code in `org-latex-impatient--get-headers`,
```emacs-lisp
(org-export-get-environment
                     (org-export-get-backend 'latex))
```
Everything seems to be pure org-mode code, so it would be a surprise for me if this is a bug from `org-latex-impatient`. Sounds more like a bug of org-mode, and the fact that it came from `org-element--cache` may suggest that inconsistent cache might be the root cause.

P.S. I have Emacs 30 (commit ec105a4) with org 9.6 (commit 86c4635), and could not reproduce the error. It would be great if you could give more precise versions of both Emacs and org-mode. Since Emacs 29 is yet to release, I assume you are on some developing channel, and you can get the exact commit id of your Emacs build with `C-h v emacs-repository-version`. As for org-mode, you might want to use `C-h f org-mode` to find the location of your org.el, and run `git log` in its directory.

**Comment by alexeyre** [comment]
Fixed by upgrading to Emacs 30.0.50.
```
GNU Emacs 30.0.50 (build 1, aarch64-apple-darwin22.2.0, NS appkit-2299.30 Version 13.1 (Build 22C65)) of 2023-01-23
```
